# Python
Ο κύριος στόχος της είναι η αναγνωσιμότητα του κώδικά της. Η ευκολία χρήσης της και το συντακτικό της επιτρέπει στους προγραμματιστές να εκφράσουν έννοιες σε λιγότερες γραμμές κώδικα απ'ότι θα ήταν δυνατόν σε άλλες γλώσσες.

Δεν απαιτεί τη δήλωση μεταβλητών και ανά πάσα στιγμή μπορούμε να εκχωρήσουμε μία τίμη σε οποιαδήποτε μεταβλητή και τύπο θελήσουμε.


# Μαθηματικές Πράξεις
Εκτός από τις βασικές μαθηματικές πράξεις, μπορούμε να χρησιμοποιήσουμε απευθείας δυνάμεις και υπόλοιπο διαίρεσεις με τη χρήση των συμβόλων ** και %

In [1]:
# πρόσθεση
print(4 + 6)

10


In [2]:
# αφαίρεση
print(10 - 8)

2


In [3]:
# πολλαπλασιασμός
print(4 * 5)

20


In [6]:
# διαίρεση
print(10 / 4)

2.5


In [7]:
# υπόλοιπο διαίρεσης
print(10 % 8)

2


In [8]:
# δύναμη
print(10 ** 2)

100


In [9]:
# αριθμητικές παραστάσεις
print(10 * 2 + (3-1)** 4)

36


# Μεταβλητές
Για να χρησιμοποιήσουμε μια νέα μεταβλητή, αρκεί να γράψουμε το όνομά της, το σύμβολο "=" και την τιμή που θέλουμε να δώσουμε ανεξάρτητα από τον τύπο (αριθμός, κείμενο κτλ)

Μπορούμε να κάνουμε τις παραπάνω πράξεις και με μεταβλητές. Ωστόσο, θα πρέπει να προσέξουμε τους τύπους που έχει η κάθε μία καθώς δε γίνεται αυτοματοποιημένος έλεγχος.

In [10]:
year = 2018
print(year)

2018


In [11]:
print(year + 1)

2019


In [12]:
print(year)

2018


Όπως βλέπουμε, η πράξεις και αλλαγές που μπορεί να γίνουν μέσα στη συνάρτηση print δεν είναι μόνιμες. Τυπώνοντας ξανά τη μεταβλητή year, συνεχίζει να έχει την παλιά τιμή που είχε. Αν όμως αντικαταστήσουμε τη μεταβλητή με νέα τιμή, η παλιά χάνεται για πάντα.

In [13]:
year = year + 1
print(year)

2019


## Είδη μεταβλητών
Υπάρχουν πολλά είδη μεταβλητών που θα συναντήσουμε. Οι βασικές είναι integer, float, string και bool.

In [14]:
print(type(year))

<class 'int'>


In [15]:
print(type(17.3))

<class 'float'>


In [16]:
year = 'δύο χιλιάδες δέκα οκτώ'
print(year)
print(type(year))

δύο χιλιάδες δέκα οκτώ
<class 'str'>


In [17]:
print(type(True))

<class 'bool'>


In [18]:
h = 'Το έτος'
w = 'είναι'
print(h,w,year)

Το έτος είναι δύο χιλιάδες δέκα οκτώ


Στη συνάρτηση print μπορούμε όπως είδαμε να δώσουμε παραπάνω από μία παράμετρο και να τυπωθεί στη σειρά με κενό ανάμεσα. Ωστόσο μπορούμε να έχουμε το ίδιο αποτέλεσμα με διαφορετικούς τρόπους.

In [19]:
print('Το έτος είναι', year)

Το έτος είναι δύο χιλιάδες δέκα οκτώ


In [22]:
print('Το έτος είναι' + ' '  + year)

Το έτος είναι δύο χιλιάδες δέκα οκτώ


Όπως είπαμε, πρέπει να προσέχουμε την τιμή που έχει η κάθε μεταβλητή. Μπορούμε να "προσθέσουμε" δύο ή περισσότερες μεταβλητές αν είναι κείμενο (string), αλλά δεν μπορούμε αν είναι κείμενο και αριθμός.

In [23]:
year = 2018
print('Το έτος είναι ' + year)

TypeError: must be str, not int

Σε αυτήν την περίπτωση, χρησιμοποιούμε είτε τη συνάρτηση str() όπου μετατρέπει κάθε αριθμό σε κείμενο ή τη συνάρτηση print με ξεχωριστούς παραμέτρους

In [24]:
print('Το έτος είναι ' + str(year))
print('Το έτος είναι', year)


Το έτος είναι 2018
Το έτος είναι 2018


In [25]:
print("Είμαι " + str(30) + " χρονών και γεννήθηκα το " + str(1988))

Είμαι 30 χρονών και γεννήθηκα το 1988


### int(), str(), float()

In [26]:
integer_number = 2018
text = '2018'
float_number = 2018.5

print(type(integer_number))
print(type(text))
print(type(float_number))

<class 'int'>
<class 'str'>
<class 'float'>


In [27]:
print(int(text))
print(float(text))
print(str(float_number))

2018
2018.0
2018.5


### Boolean
Μία από τις βασικές μεταβλητές είναι και η Bool. Οι τιμές που μπορεί να πάρει είναι True & False. Προσοχή, πάντα με κεφαλαίο το πρώτο γράμμα.

In [28]:
a = True
b = False

print(a)
print(type(a))

True
<class 'bool'>


In [29]:
wrong_bool = true

NameError: name 'true' is not defined

Μπορούμε να "συγκρίνουμε" μεταβλητές και το αποτέλεσμα που θα πάρουμε να είναι Boolean. Οι διαθέσιμοι "operators" είναι

* == (ίσο)
* != (διαφορετικό)
* \> (μεγαλύτερο)
* \>= (μεγαλύτερο ή ίσο)
* < (μικρότερο)
* <= (μικρότερο ή ίσο)
* in (υπάρχει μέσα σε λίστα)

In [30]:
# ισότητα
print(3 == 5)
print(3 == 3)

False
True


In [31]:
# ανισότητα
print(3 != 5)
print(3 != 3)

True
False


In [32]:
# μεγαλύτερο
print(3 > 5)
print(5 > 3)
print(5 >= 5)

False
True
True


In [33]:
# υπάρχει μέσα σε λίστα
3 in [1,2,3,4,5,6]

True

Για περισσότερες πληροφορίες σχετικά με τους πίνακες αληθείας δείτε εδώ https://en.wikipedia.org/wiki/Truth_table

# Λίστες

Οι λίστες της Python έχουν δυναμικό μέγεθος και τα αντικείμενά τους είναι οποιουδήποτε τύπου. Μία λίστα ξεκινάει και τελειώνει με τα σύμβολα [ και ] και τα αντικείμενα που έχει χωρίζονται με κόμμα.

In [34]:
empty_list = []
print(empty_list)

[]


In [37]:
my_list = [2017, 2018, 'μπανάνα', 'data', True]
print(my_list)

[2017, 2018, '"μπανάνα"', 'data', True]


In [36]:
print(type(my_list))

<class 'list'>


In [38]:
year = 2018
month = "September"
day = "Saturday"

today = ["year", year, "month", month, "day", day]
print(today)

['year', 2018, 'month', 'September', 'day', 'Saturday']


Μπορούμε να βρούμε το μέγεθος της λίστας (πόσα αντικείμενα περιέχει) χρησιμοποιώντας τη συνάρτηση len(). 

In [39]:
len(today)

6

### Αρίθμηση λίστας

Μπορούμε να επιστρέψουμε όποιο στοιχείο θέλουμε από την λίστα χρησιμοποιόντας τη θέση που βρίσκεται σε αυτήν. Προσοχή!!! Η αρίθμηση ξεκινάει από 0 και όχι από 1

In [40]:
print(today[0])
print(today[1])

year
2018


Και αντίστροφα ;) Μπορούμε ανάποδα ξεκινώντας από το -1

In [41]:
print(today[-1])
print(today[-4])

Saturday
month


### Προσθήκη και αφαίρεση στοιχείων από τη λίστα

Μπορούμε να προσθέσουμε στο τέλος της λίστας ένα στοιχείο με την εντολή my_list.append(4) ή να αφαιρέσουμε ένα με my_list.remove('μπανάνα')

In [42]:
my_list.append(10)
print(my_list)
back_up = my_list[:]

[2017, 2018, '"μπανάνα"', 'data', True, 10]


In [43]:
my_list.remove(10)
print(my_list)

[2017, 2018, '"μπανάνα"', 'data', True]


In [44]:
print(my_list)

[2017, 2018, '"μπανάνα"', 'data', True]


### Αντικατάσταση στοιχείων στη λίστα

Αν θέλουμε να αντικαταστήσουμε ένα στοιχείο της λίστας με ένα καινούργιο, μπορούμε απλά να το δηλώσουμε όπως κάναμε στις μεταβλητές, χρησιμοποιώντας τη θέση που βρίσκεται

In [47]:
my_list[2] = "μήλο"
print(my_list)

[2017, 2018, 'μήλο', 'data', True]


### Ταξινόμιση
Για ταξινόμιση μιας λίστας, μπορούμε να χρησιμοποιήσουμε τη συνάρτηση sorted()

In [48]:
numbers = [1,5,2,6,7,3,2,4,6,8,9,3,2,6,8,9,5,4,3,4,6,8,9,5,4]

In [50]:
# αύξουσα σειρά
print(sorted(numbers))
print(numbers)

[1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 6, 6, 6, 6, 7, 8, 8, 8, 9, 9, 9]
[1, 5, 2, 6, 7, 3, 2, 4, 6, 8, 9, 3, 2, 6, 8, 9, 5, 4, 3, 4, 6, 8, 9, 5, 4]


In [51]:
# φθίνουσα σειρά
print(sorted(numbers, reverse = True))

[9, 9, 9, 8, 8, 8, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 4, 3, 3, 3, 2, 2, 2, 1]


### List methods
index(), count() και reverse()

In [52]:
# επιστρέφει τη θέση του πρώτου στοιχείου με την τιμή που θέλουμε
print(numbers.index(7))

4


In [53]:
# επιστρέφει το σύνολο των στοιχείων με την τιμή που θέλουμε
print(numbers.count(5))

3


In [54]:
# επιστρέφει τη λίστα με αντίστροφη σειρά
# προσοχή. Καλώντας τη μέθοδο reverse, η αρχική μας λίστα χάνεται
numbers.reverse()
print(numbers)

[4, 5, 9, 8, 6, 4, 3, 4, 5, 9, 8, 6, 2, 3, 9, 8, 6, 4, 2, 3, 7, 6, 2, 5, 1]


### Άσκηση

1. Δημιουργήστε 2 κενές λίστες. Η πρώτη να λέγεται cities και η δεύτερη populations
2. Με χρήση της append προσθέστε 5 πόλεις της Ελλάδας
3. Αφού βρείτε τον πληθυσμό των πόλεων από τη Wikipedia, συμπληρώστε και τη δεύτερη λίστα με την ίδια σειρά
4. Τυπώστε τη δεύτερη πόλη μαζί με τον πληθυσμό της στη μορφή "Καβάλα: 80000"

In [56]:
cities = []
populations = []
cities.extend(['Kavala', 'Thessaloniki', 'Athens', 'Volos', 'Kozani'])
populations.extend([60000, 1000000, 5500000, 40000, 50000])
print(cities[1]+": " + str(populations[1]))
print(cities[1],":", populations[1])
print("Η πόλη είναι η {}: {}".format(cities[1],populations[1]))

Thessaloniki: 1000000
Thessaloniki : 1000000
Η πόλη είναι η Thessaloniki: 1000000


### Υπολίστες

Μπορούμε να επιλέξουμε οποιοδήποτε κομμάτι της λίστας. Όχι μόνο ένα στοιχείο όπως προηγουμένως, αλλά πολλά μαζί.

In [57]:
my_list = [2017, 2018, 'μπανάνα', 'data', True, 'true', 10, 10]
print(my_list)
my_list[0:3]

[2017, 2018, 'μπανάνα', 'data', True, 'true', 10, 10]


[2017, 2018, 'μπανάνα']

#### Προσοχή!

Το τελευταίο στοιχείο λείπει! Έχουν επιστραφεί τα στοιχεία στις θέσεις 0, 1 και 2.

In [58]:
my_list[:3]

[2017, 2018, 'μπανάνα']

In [59]:
my_list[3:]

['data', True, 'true', 10, 10]

In [61]:
my_list[-3:-1]

['true', 10]

### Λίστες μέσα σε λίστες

In [62]:
list_of_lists = [[1,2,3],[4,5,6],[7,8,9]]
print(list_of_lists)
list_of_lists[0]

[[1, 2, 3], [4, 5, 6], [7, 8, 9]]


[1, 2, 3]

In [63]:
type(list_of_lists)

list

In [64]:
list_of_lists[0][1]

2

### Split & Join

In [65]:
years = '2017,2018,2019,2020'
years.split(",")

['2017', '2018', '2019', '2020']

In [67]:
t = 'Τα αρχεία τα ανοίγουμε με την εντολή open(file, "r"). Το "r" δηλώνει ότι πρόκειται μόνο για read. Αν είχε "w" θα είχαμε δικαιώματα για να γράψουμε και στο αρχείο νέα δεδομένα. Αφού ανοίξουμε το αρχείο, για να μπορούμε να το επεξεργαστούμε, πρέπει να το "διαβάσουμε"'
another_list = t.split(' ')
print(another_list)
len(another_list)

['Τα', 'αρχεία', 'τα', 'ανοίγουμε', 'με', 'την', 'εντολή', 'open(file,', '"r").', 'Το', '"r"', 'δηλώνει', 'ότι', 'πρόκειται', 'μόνο', 'για', 'read.', 'Αν', 'είχε', '"w"', 'θα', 'είχαμε', 'δικαιώματα', 'για', 'να', 'γράψουμε', 'και', 'στο', 'αρχείο', 'νέα', 'δεδομένα.', 'Αφού', 'ανοίξουμε', 'το', 'αρχείο,', 'για', 'να', 'μπορούμε', 'να', 'το', 'επεξεργαστούμε,', 'πρέπει', 'να', 'το', '"διαβάσουμε"']


45

In [68]:
another_list = ['3','5','True']
'-'.join(another_list)


'3-5-True'

### Άνοιγμα αρχείων

Τα αρχεία τα ανοίγουμε με την εντολή open(file, "r"). Το "r" δηλώνει ότι πρόκειται μόνο για read. Αν είχε "w" θα είχαμε δικαιώματα για να γράψουμε και στο αρχείο νέα δεδομένα. Αφού ανοίξουμε το αρχείο, για να μπορούμε να το επεξεργαστούμε, πρέπει να το "διαβάσουμε"

In [70]:
f = open("crime_rates.csv", "r")
data = f.read()
print(data)

Albuquerque,749
Anaheim,371
Anchorage,828
Arlington,503
Atlanta,1379
Aurora,425
Austin,408
Bakersfield,542
Baltimore,1405
Boston,835
Buffalo,1288
Charlotte-Mecklenburg,647
Cincinnati,974
Cleveland,1383
Colorado Springs,455
Corpus Christi,658
Dallas,675
Denver,615
Detroit,2122
El Paso,423
Fort Wayne,362
Fort Worth,587
Fresno,543
Greensboro,563
Henderson,168
Houston,992
Indianapolis,1185
Jacksonville,617
Jersey City,734
Kansas City,1263
Las Vegas,784
Lexington,352
Lincoln,397
Long Beach,575
Los Angeles,481
Louisville Metro,598
Memphis,1750
Mesa,399
Miami,1172
Milwaukee,1294
Minneapolis,992
Mobile,522
Nashville,1216
New Orleans,815
New York,639
Newark,1154
Oakland,1993
Oklahoma City,919
Omaha,594
Philadelphia,1160
Phoenix,636
Pittsburgh,752
Plano,130
Portland,517
Raleigh,423
Riverside,443
Sacramento,738
San Antonio,503
San Diego,413
San Francisco,704
San Jose,363
Santa Ana,401
Seattle,597
St. Louis,1776
St. Paul,722
Stockton,1548
Tampa,616
Toledo,1171
Tucson,724
Tulsa,990
Virginia Beach,1

### New line
Στην Python δηλώνουμε νέα γραμμή στο κείμενο με τους χαρακτήρες \n . Προσοχή. Οι χαρακτήρες αυτοί δε θα τυπωθούν όταν καλέσουμε την print

In [71]:
print('a \nb')

a 
b


In [72]:
rows = data.split('\n')
print(rows[0:5])

['Albuquerque,749', 'Anaheim,371', 'Anchorage,828', 'Arlington,503', 'Atlanta,1379']


### For Loop

In [73]:
ten_rows = rows[0:10]
for row in ten_rows:
    print(row)

Albuquerque,749
Anaheim,371
Anchorage,828
Arlington,503
Atlanta,1379
Aurora,425
Austin,408
Bakersfield,542
Baltimore,1405
Boston,835


### Range

Με την εντολή range μπορώ να έχω μία λίστα αριθμών με το εύρος και βήμα που ορίζω εγώ

In [74]:
for i in range(10):
    print(i)

0
1
2
3
4
5
6
7
8
9


In [75]:
for j in range(5,10):
    print(j,end=',')

5,6,7,8,9,

In [76]:
for k in range(0,10,2):
    print(k,end=',')

0,2,4,6,8,



### Άσκηση

Να βρεθεί ο 15ος αριθμός Fibonacci αν γνωρίζετε ότι ο 1ος και 2ος είναι ίσος με 1 και κάθε επόμενος προκύπτει από το άθροισμα των δύο προηγούμενων

In [77]:
a1 = 1
a2 = 1
for n in range(13):
    temp = a1 + a2
    a1 = a2
    a2 = temp
print(a2)

610


### Και άλλες λίστες...

In [78]:
crimes = []
for row in rows:
    split_list = row.split(',')
    crimes.append(split_list)
print(crimes[:5])

[['Albuquerque', '749'], ['Anaheim', '371'], ['Anchorage', '828'], ['Arlington', '503'], ['Atlanta', '1379']]


Η υπολίστα με την πόλη Atlanta είναι στη θέση 4. Οπότε αν θέλουμε να τυπώσουμε την τιμή 1379, πρέπει πρώτα να πάρουμε την υπολίστα και μετά το δεύτερο στοιχείο από αυτήν.

In [79]:
print(crimes[4][0])
print(crimes[4][1])

Atlanta
1379


In [80]:
int(crimes[4][1]) > int(crimes[3][1])

True

### If statements

Όταν θέλω να συγκρίνω δύο μεταβλητές, χρησιμοποιώ την If, else

In [81]:
my_favourite_fruit = "apple"

if my_favourite_fruit == "orange":
    print('You are lucky')

In [82]:
my_favourite_fruit = "apple"

if my_favourite_fruit == "orange":
    print('You are lucky')
else:
    print('Damn...')

Damn...


In [84]:
grade = 71
if grade > 90 and grade <= 100:
    print('Well done')
elif grade > 80:
    print('Μπορείς και καλύτερα')
elif grade > 70:
    print('Θες βελτίωση')
else:
    print('Δυστυχώς ξανά προσπάθησε')

Θες βελτίωση


### Άσκηση

Να βρεθεί η πόλη με την μεγαλύτερη εγκληματικότητα. Προσοχή στους τύπους μεταβλητών!

In [85]:
crime_rate = 0
crime_city = ''
for city in crimes:
    if int(city[1]) > crime_rate:
        crime_city = city[0]
        crime_rate = int(city[1])
        
print(crime_city, crime_rate)

Detroit 2122


### Άσκηση

1. Να βρεθεί dataset με όλες τις πόλεις και τον πληθυσμό τους
2. Να βρεθεί η πόλη με τον μικρότερο πληθυσμό

In [86]:
g = open("populations.csv", "r", encoding="utf8")
data = g.read().split('\n')
data[:10]
smallest_city = ''
smallest_pop = False
for row in data[1:-2]:
    city = row.split(',')
    if float(city[4]) < smallest_pop:
        smallest_pop = float(city[4])
        smallest_city = city[0]
print(smallest_city, smallest_pop)

Fortin Falcon -99.0


## Πικρή Αλήθεια

Το μεγαλύτερο ποσοστό των dataset περιέχουν λάθη που πρέπει εμείς να βρούμε και να αποφασίσουμε τι θα τα κάνουμε :(

## Λεξικά

Μια άλλη μορφή δεδομένων είναι τα Λεξικά (Dictionaries) και χρησιμοποιούνται για να βρίσκουμε εύκολα ένα αντικείμενο με βάση κάποιο κλειδί

In [91]:
dictionary = {}
dictionary[1] = 'one'
dictionary['one'] = 1
dictionary['a_list'] = [1,2,3,4]
dictionary['a_boolean'] = True
dictionary['a_number'] = 5
print(dictionary)
print(dictionary['a_boolean'])

{1: 'one', 'one': 1, 'a_list': [1, 2, 3, 4], 'a_boolean': True, 'a_number': 5}
True


In [89]:
print(dictionary['one'])

1


In [92]:
dictionary['a_number']

5

In [93]:
dictionary['nothing']

KeyError: 'nothing'

In [94]:
print(dictionary.keys())

dict_keys([1, 'one', 'a_list', 'a_boolean', 'a_number'])


In [95]:
f = open("crime_rates.csv", "r")
data = f.read().split('\n')
cities = {}
for row in data:
    city = row.split(',')
    cities[city[0]] = int(city[1])

print(cities)

{'Albuquerque': 749, 'Anaheim': 371, 'Anchorage': 828, 'Arlington': 503, 'Atlanta': 1379, 'Aurora': 425, 'Austin': 408, 'Bakersfield': 542, 'Baltimore': 1405, 'Boston': 835, 'Buffalo': 1288, 'Charlotte-Mecklenburg': 647, 'Cincinnati': 974, 'Cleveland': 1383, 'Colorado Springs': 455, 'Corpus Christi': 658, 'Dallas': 675, 'Denver': 615, 'Detroit': 2122, 'El Paso': 423, 'Fort Wayne': 362, 'Fort Worth': 587, 'Fresno': 543, 'Greensboro': 563, 'Henderson': 168, 'Houston': 992, 'Indianapolis': 1185, 'Jacksonville': 617, 'Jersey City': 734, 'Kansas City': 1263, 'Las Vegas': 784, 'Lexington': 352, 'Lincoln': 397, 'Long Beach': 575, 'Los Angeles': 481, 'Louisville Metro': 598, 'Memphis': 1750, 'Mesa': 399, 'Miami': 1172, 'Milwaukee': 1294, 'Minneapolis': 992, 'Mobile': 522, 'Nashville': 1216, 'New Orleans': 815, 'New York': 639, 'Newark': 1154, 'Oakland': 1993, 'Oklahoma City': 919, 'Omaha': 594, 'Philadelphia': 1160, 'Phoenix': 636, 'Pittsburgh': 752, 'Plano': 130, 'Portland': 517, 'Raleigh': 4

In [96]:
cities['Thessaloniki'] = 100

In [97]:
cities

{'Albuquerque': 749,
 'Anaheim': 371,
 'Anchorage': 828,
 'Arlington': 503,
 'Atlanta': 1379,
 'Aurora': 425,
 'Austin': 408,
 'Bakersfield': 542,
 'Baltimore': 1405,
 'Boston': 835,
 'Buffalo': 1288,
 'Charlotte-Mecklenburg': 647,
 'Cincinnati': 974,
 'Cleveland': 1383,
 'Colorado Springs': 455,
 'Corpus Christi': 658,
 'Dallas': 675,
 'Denver': 615,
 'Detroit': 2122,
 'El Paso': 423,
 'Fort Wayne': 362,
 'Fort Worth': 587,
 'Fresno': 543,
 'Greensboro': 563,
 'Henderson': 168,
 'Houston': 992,
 'Indianapolis': 1185,
 'Jacksonville': 617,
 'Jersey City': 734,
 'Kansas City': 1263,
 'Las Vegas': 784,
 'Lexington': 352,
 'Lincoln': 397,
 'Long Beach': 575,
 'Los Angeles': 481,
 'Louisville Metro': 598,
 'Memphis': 1750,
 'Mesa': 399,
 'Miami': 1172,
 'Milwaukee': 1294,
 'Minneapolis': 992,
 'Mobile': 522,
 'Nashville': 1216,
 'New Orleans': 815,
 'New York': 639,
 'Newark': 1154,
 'Oakland': 1993,
 'Oklahoma City': 919,
 'Omaha': 594,
 'Philadelphia': 1160,
 'Phoenix': 636,
 'Pittsburgh

### Αναζήτηση σε λεξικά
Για να δούμε αν υπάρχει κάποιο κλειδί μέσα στο λεξικό αρκεί να χρησιμοποιήσουμε τον operator in

In [98]:
'Oklahoma City' in cities

True

In [99]:
'Athens' in cities

False

### Διαγραφή κλειδιού
Για τη διαγραφή ενός κλειδιού αρκεί να καλέσουμε τη συνάρτηση del() στο λεξικό και το κλειδί που θέλουμε να διαγράψουμε

In [100]:
del(cities['Thessaloniki'])

### Ταξινόμιση λεξικών
Για να κάνουμε ταξινόμιση, πρέπει πρώτα να μετατρέψουμε το λεξικό σε λίστα με τη χρήση της μεθόδου items(). Μετά να δηλώσουμε στην παράμετρο key ότι θέλουμε να γίνει με βάση την τιμή και όχι το κλειδή με χρήση της lamda
sorted(cities.items(), key=lambda x:x[1], reverse=True)

In [101]:
sorted(cities.items(), key=lambda x:x[1], reverse=True)

[('Detroit', 2122),
 ('Oakland', 1993),
 ('St. Louis', 1776),
 ('Memphis', 1750),
 ('Stockton', 1548),
 ('Baltimore', 1405),
 ('Cleveland', 1383),
 ('Atlanta', 1379),
 ('Milwaukee', 1294),
 ('Buffalo', 1288),
 ('Kansas City', 1263),
 ('Nashville', 1216),
 ('Indianapolis', 1185),
 ('Washington', 1177),
 ('Miami', 1172),
 ('Toledo', 1171),
 ('Philadelphia', 1160),
 ('Newark', 1154),
 ('Houston', 992),
 ('Minneapolis', 992),
 ('Tulsa', 990),
 ('Cincinnati', 974),
 ('Oklahoma City', 919),
 ('Boston', 835),
 ('Anchorage', 828),
 ('New Orleans', 815),
 ('Las Vegas', 784),
 ('Pittsburgh', 752),
 ('Albuquerque', 749),
 ('Wichita', 742),
 ('Sacramento', 738),
 ('Jersey City', 734),
 ('Tucson', 724),
 ('St. Paul', 722),
 ('San Francisco', 704),
 ('Dallas', 675),
 ('Corpus Christi', 658),
 ('Charlotte-Mecklenburg', 647),
 ('New York', 639),
 ('Phoenix', 636),
 ('Jacksonville', 617),
 ('Tampa', 616),
 ('Denver', 615),
 ('Louisville Metro', 598),
 ('Seattle', 597),
 ('Omaha', 594),
 ('Fort Worth', 

### Λεξικά μέσα σε λεξικά

In [102]:
countries = { 'Greece': { 'capital':'Athens', 'population':10750000 },
           'Sweden': { 'capital':'Stockholm', 'population':9903000000 },
           'Denmark': { 'capital':'Copenhagen', 'population':5731000000 }}

In [103]:
print(countries)

{'Greece': {'capital': 'Athens', 'population': 10750000}, 'Sweden': {'capital': 'Stockholm', 'population': 9903000000}, 'Denmark': {'capital': 'Copenhagen', 'population': 5731000000}}


In [104]:
print(countries['Greece'])

{'capital': 'Athens', 'population': 10750000}


In [105]:
print(countries['Greece']['population'])

10750000


## Συναρτήσεις

Πάντα προσπαθούμε να σπάσουμε τον κώδικά μας σε συναρτήσεις. Όσο μεγαλώνει το project, θα γίνεται όλο και πιο δύσκολη η συντήρηση και οι αλλαγές που θα θέλουμε να κάνουμε.

In [106]:
def my_function(a, b = 5):
    if a > 0:
        c = a + b
    else:
        c = -3 * a + b
    return c

r1 = my_function(5)
r2 = my_function(-5)
r3 = my_function(5,10)
print(r1, r2, r3)

10 20 15


### Άσκηση

Να γραφτεί συνάρτηση που θα δέχεται λίστα και θα επιστρέφει το μέγιστο. Για ευκολία, θεωρούμε ότι η λίστα θα περιέχει μόνο αριθμούς.

In [107]:
def max_of_list(my_list):
    return max(my_list)

max_of_list([0,1,2,3,4,5,6,7,8,100])

100

In [108]:
def my_print():
    print('Καλημέρα')

my_print()

Καλημέρα


In [ ]:
def my_message(message="Καλημέρα"):
    print(message)
my_message()

### Βιβλιοθήκες
Για να χρησιμοποιήσουμε κάποια βιβλιοθήκη της Python, αρκεί να την εισάγουμε με την εντολή import στην αρχή του κώδικα

In [109]:
import math
print(math.pi)

3.141592653589793


In [110]:
print(math.pow(3,2))

9.0


Υπάρχει η δυνατότητα να εισάγουμε μέρος της βιβλιοθήκης και όχι ολόκληρη

In [ ]:
from math import pow
print(pow(3,2))

Ή να δώσουμε nickname στη βιβλιοθήκη βάζοντας στο τέλος το as και το όνομα που θέλουμε

In [111]:
import math as mathimatika
print(mathimatika.pi)

3.141592653589793
